### PR TITLE
Dashboard Update

### PR DESCRIPTION
**Pull request by c-schwarze**

### CODE REVIEW

File: app/dashboards/__init__.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    def is_partial_string_in_array(partial_string, array):
+        for string in array:
+            if partial_string in string:
+                return True
+        return False
+
+    # todo: need to cache this
+    def build_event_feed(url):
+
+        response = urllib2.urlopen(url)
```
Comment (e-jameson) [inline app/dashboards/__init__.py]: Currently its using both `requests` and `urllib2`, can we use only one? I'd prefer `requests`

File: app/dashboards/__init__.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+        event_list = []
+        for event in form_xml.findall('.//event'):
+            if event:
+                event_list.append(event)
+
+        event_list = sorted(event_list, key=lambda k: int(k.find('.//start-date').text))
+
+        try:
+            event_list = event_list[:int(event_feed_num_events)]
```
Comment (e-jameson) [inline app/dashboards/__init__.py]: can you comment this?

File: app/templates/base.html [hunk 3]
```diff
@@ -61,7 +61,10 @@
                     {% include 'admin/admin_user_bar.html' %}
 
                     {% if request.path == '/' or request.path == '/home' %}
-                        {% include 'modules/dashboard.html' %}
+                        <div class="dashboard">
+                            <h2 class="gamma  mb-4">Hi Caleb, welcome back!</h2>
```
Comment (e-jameson) [inline app/templates/base.html]: Should be able to load name from the session

File: app/dashboards/__init__.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    dashboards = []
+
+    # staff or faculty
+    if is_partial_string_in_array('STAFF', session['user_roles']) or is_partial_string_in_array('FACULTY', session['user_roles']):
+        # todo: make sure this handles errors properly
+        leave_balances_url = 'https://wsapi.bethel.edu/portal/leave-balances'
+        leave_balances_response = requests.post(leave_balances_url, data={'username': session['username']})
+        leave_balances_data = leave_balances_response.json().get('leave_balances').get('0')
+
+        event_feed = build_event_feed('https://www.bethel.edu/_blink-content/portal-content/event-feeds/employee-holiday-portal-event-feed.xml')
```
Comment (e-jameson) [inline app/dashboards/__init__.py]: these URLs should come from the `config` file I think

File: app/dashboards/__init__.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        # event options
+        event_options = form_xml.find('.//event-feed-options')
+
+        # todo: if these fail, the page doesn't load. we should fix this
+        event_feed_title = event_options.find('title').text or None
+        event_feed_num_events = event_options.find('num-events').text or None
+        event_feed_add_button = event_options.find('add-button').text or None
+        event_feed_button_text = event_options.find('button-text').text or None
+        event_feed_more_events_link = event_options.find('more-events-link').text or None
+
```
Comment (phg49389) [inline app/dashboards/__init__.py]: In the for loop below, could it be changed to list comprehension? A la 
```
event_list = [event for event in form_xml.finall('.//event') if event]
```

File: app/dashboards/__init__.py [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+        event_list = []
+        for event in form_xml.findall('.//event'):
+            if event:
+                event_list.append(event)
+
+        event_list = sorted(event_list, key=lambda k: int(k.find('.//start-date').text))
+
```
Comment (phg49389) [inline app/dashboards/__init__.py]: And then that list comprehension could be wrapped in this sort pretty easily

File: app/dashboards/__init__.py (hunk 5 above)
Comment (c-schwarze) [inline app/dashboards/__init__.py]: This works, but is that actually better? I think it loses a lot of readability. Thoughts? @e-jameson @phg49389

Comment (e-jameson) [inline app/dashboards/__init__.py]: Don't have a strong feeling either way about this one.

File: app/dashboards/__init__.py (hunk 6 above)
Comment (e-jameson) [inline app/dashboards/__init__.py]: Not a fan of wrapping a comprehension in a `sorted` / `lambda`

File: app/dashboards/__init__.py (hunk 5 above)
Comment (c-schwarze) [inline app/dashboards/__init__.py]: I'll keep it as is for now and add a comment.

**Pull request merged**